internal/metrics/metrics.go

```diff
@@ -0,0 +1,38 @@
+package metrics
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// https://prometheus.io/docs/practices/naming/
+var (
+	ClickHouseBatchSize = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "jaeger_clickhouse_batch_size",
+		Help: "Maximum size of a batch written to clickhouse",
+	})
+	ClickHouseFlushInterval = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "jaeger_clickhouse_flush_interval_seconds",
+		Help: "Maximum wait interval before a batch written to clickhouse",
+	})
+	NumWritesWithBatchSize = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "jaeger_clickhouse_writes_with_batch_size_total",
+		Help: "Number of clickhouse writes due to batch size criteria",
+	})
+	NumWritesWithFlushInterval = prometheus.NewCounter(prometheus.CounterOpts{
+		Name: "jaeger_clickhouse_writes_with_flush_interval_total",
+		Help: "Number of clickhouse writes due to flush interval criteria",
+	})
+)
+
+var registerMetrics sync.Once
+
+func Register() {
```

Comment:
Can we avoid the global state and pass p8s instance/metric registry to `NewWriter` and register it there?